nionswift_plugin/usim/InstrumentDevice.py

```diff
@@ -356,6 +356,22 @@ def __init__(self, instrument_id: str):
         self.live_probe_position = None
         theta = self.__tv_pixel_angle * self.__ronchigram_shape.height / 2  # half angle on camera
         self.__aberrations_controller = AberrationsController(self.__ronchigram_shape[0], self.__ronchigram_shape[1], theta, self.__max_defocus, self.__defocus_m)
+        self.__sequence_progress = 0
+        self.__lock = threading.Lock()
```

Comment:
Is this lock necessary? All of the methods look more or less atomic.

Reply:
Probably it is not really necessary, I just put it in there to be extra sure nothing goes wrong since sequence_progress is a very critical counter.